internal/api/middleware/timer.go

```diff
@@ -0,0 +1,43 @@
+package middleware
+
+import (
+	"net/http"
+	"time"
+
+	"regexp"
+	"strings"
+
+	"github.com/go-chi/chi"
+	"github.com/go-kit/kit/metrics"
+	"github.com/go-kit/kit/metrics/dogstatsd"
+)
+
+func getHist(r *http.Request, t *dogstatsd.Timing) metrics.Histogram {
+	str := strings.ReplaceAll(chi.RouteContext(r.Context()).RoutePattern(), "/", ".")
+	regx := regexp.MustCompile(`[{}]`)
+	if regx.MatchString(str) {
```

Comment:
I don't think we need to check for matches first. If there are no matches, then `ReplaceAllString` does nothing anyways.